cmake/Common/CompilerArmClangCommon.cmake

```diff
@@ -140,6 +140,8 @@ function(compiler_set_cmse_output TARGET FILE_PATH)
 	set_property(TARGET ${TARGET} APPEND_STRING PROPERTY LINK_FLAGS " --import_cmse_lib_out=${FILE_PATH}")
 	#Tell cmake cmse output is a generated object file.
 	SET_SOURCE_FILES_PROPERTIES("${FILE_PATH}" PROPERTIES EXTERNAL_OBJECT true GENERATED true)
+	#Dummy command to add BYPRODUCTS
+	add_custom_command(TARGET ${TARGET} POST_BUILD COMMAND "" BYPRODUCTS ${FILE_PATH})
```

Comment:
Something is not right here.
The `${FILE_PATH}` already have the generated property set.

But to me, the naming indicates this is a folder and not a file, and that is something different.

Will take a closer look, but if you can provide some more information on this, it will be helpful.

Reply:
FILE_PATH is S_VENEER_FILE

	set(VENEER_NAME ${_MY_PARAMS_VENEER_NAME}${_MY_PARAMS_POSTFIX}.o)
	...  
	set(S_VENEER_FILE "${CMAKE_CURRENT_BINARY_DIR}/${VENEER_NAME}")
	compiler_set_cmse_output(${EXE_NAME} "${S_VENEER_FILE}")

This is the only place `compiler_set_cmse_output` is called.

Ninja was failing without these lines.
Something like

    s_veneers.o missing and no rule to make it

Reply:
I did some more investigations on this.
The line above `SET_SOURCE_FILES_PROPERTIES("${FILE_PATH}" PROPERTIES EXTERNAL_OBJECT true GENERATED true)` was suppose to also fix this, but seems the ninja and make generator behaves slightly different.

But your suggestion should in theory solve this.
I did some examination of the CMake code, and the way properties are set and link dependencies makes it a bit tricky, and we don't want to make to many mods to TF-M build system.

We could in theory remove the mentioned line, but as it does also not harm, let's keep out changes to a minimum, unless we decide to send a patch upstream.